File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/carts/domain/repository/CartRepository.java
package com.tarbadev.witchcraft.carts.domain.repository;

import com.tarbadev.witchcraft.carts.domain.entity.Cart;

import java.util.List;

public interface CartRepository {
  List<Cart> findAll();
  Cart save(Cart cart);
  Cart findById(Integer id);
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/carts/domain/repository/CartRepository.kt
package com.tarbadev.witchcraft.carts.domain.repository

import com.tarbadev.witchcraft.carts.domain.entity.Cart

interface CartRepository {
    fun findAll(): List<Cart>
    fun save(cart: Cart): Cart
    fun findById(id: Int): Cart?
    fun delete(id: Int)
}
